addons/website_mail/controllers/main.py

```diff
@@ -27,7 +33,9 @@ def _message_post_helper(res_model='', res_id=None, message='', token='', token_
     """
     record = request.env[res_model].browse(res_id)
     author_id = request.env.user.partner_id.id if request.env.user.partner_id else False
-    if token and record and token == getattr(record.sudo(), token_field, None):
+
+    access_as_sudo = _special_access_object(res_model, res_id, token=token, token_field=token_field)
```

Comment:
Shouldn't we do something if the token is not correct ? If I understand well the record will not be accessed as sudo, but maybe it could just avoid doing stuff if a wrong token is given.